cep-sigstore-predicate.md

```diff
@@ -0,0 +1,303 @@
+      trailing slashes.
+
+An example of a compliant statement is provided below:
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v0.1",
+    "subject": [{
+        "name": "file-name-0.0.1-h123456_5.conda",
+        "digest": {"sha256": "01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b"},
+    }],
+    "predicateType": "https://schemas.conda.org/attestations/publish/v1",
+    "predicate": {
+        "targetChannel": "https://prefix.dev/conda-forge",
+    }
+}
+```
+
+### Signing and distributing
+
+This CEP recommends the following signing process:
+
+1. The signer (i.e. Alice or Alice's trusted machine identity) uses a
+   [Sigstore]-compatible client to generate an ephemeral keypair and bind it to
+   their identity via a public certificate.
+2. The signer generates an in-toto statement as described above, and
+   produces an attestation by signing that statement with their ephemeral
+   private key.
+3. The signer uploads their attestation to the Sigstore transparency log
+   as a [DSSE] envelope.
+4. The signer produces a [Sigstore bundle] containing their certificate,
+   attestation, and transparency log inclusion proof.
+
+Each of these steps is performed transparently by a Sigstore client like
+[sigstore-python], except for step (2) as it concerns the specific
+layout of the signed-over statement.
+
+The result of this process is a single Sigstore bundle, which can be
+distributed alongside the conda package or otherwise made discoverable.
+
+This CEP does not proscribe a distribution mechanism; see
+[Future work](#future-work).
+
+### Verifying
+
+This CEP recommends the following verification process:
+
+1. The verifier retrieves Alice's conda package and associated
+   Sigstore bundle.
+1. The verifier performs a standard Sigstore verification process against
+   the bundle, using Alice's identity (or machine identity) as the
+   signing identity. This process produces a verified in-toto statement.
+
+   This step requires the verifier to establish trust in the identity
+   being verified against.
+
+   Exact mechanisms for establishing this trust are
+   outside the scope of this CEP; see [Future work](#future-work).
+
+1. The verifier checks the in-toto statement for consistency against their
+   ground truth:
+
+   - The `predicateType` field **MUST** be `https://schemas.conda.org/attestations/publish/v1`.
+   - The `subject[0].name` field **MUST** match the filename of the conda package.
+   - The `subject[0].digest` field **MUST** match the SHA256 hash of the conda
+     package.
+   - The `predicate.targetChannel` field **SHOULD** match the channel that
+     the package was retrieved from, if `predicate` is present. However, the
+     verifier **MAY** choose to allow a channel mismatch, e.g. if the known
+     context is a mirroring context (where the conda package was originally
+     published to a different channel, but is now being consumed from
+     a mirror).
```

Comment:
Possibly not worth addressing in this version of the spec, however I imagine a scenario:

User creates a mirror that pulls in a few specific packages or channels together. User *wants* to validate publish attestations against all the packages as they use them (not at mirror time) -- how can a user identify that YES this package listed as published to `cf-staging`, and this other package to `somesecretorg` are valid, unless they blindly trust the set of `[null, cf-staging, somesecretorg]` to be valid for any package they use, regardless of where it truly came from.